llvm/lib/Support/Windows/Path.inc

```diff
@@ -479,6 +479,10 @@ static std::error_code rename_internal(HANDLE FromHandle, const Twine &To,
     unsigned Error = GetLastError();
     if (Error == ERROR_SUCCESS)
       Error = ERROR_CALL_NOT_IMPLEMENTED; // Wine doesn't always set error code.
+    else if (Error == ERROR_INVALID_PARAMETER) {
+      // In a Docker container, the call fails for paths in mounted volumes.
```

Comment:
I think that we should make this more generic as this can fail on non-NTFS file systems (e.g. FAT32) even without containerization.

```suggestion
      // This operation can fail on file systems that do not properly support the extended POSIX semantics for renames.
```

Reply:
I don't know if these scenarios would fail with `ERROR_INVALID_PARAMETER` though.

Reply:
The error is raised by the file system driver if it does not support the new flag `FileRenameInfoEx`.  This was a known limitation of the original change, where the expectation was, who would ever do this on a non-native NTFS file system?!  Oh right, I would 🤦‍♂️

Reply:
Applied change, thanks!